### ISSUE TITLE
Improve Reql errors

### ISSUE THREAD

**Issue by neumino** [rethinkdb/rethinkdb#559]
It would be really nice to have a more precise type for a reql error.

Like when a database already exists, it would be nice if I could do a simple check like
`err.type is 'database exists'` instead of 

```
/Database `.*` already exists.\nFailed assertion: status == METADATA_ERR_NONE\n... (length: (\d+))/.test(err.type)
```

**Comment by techdragon** [comment]
I'm currently trying to Integrate RethinkDB with Django, and I just ran into a logical condition where I need to check for multiple conditions in a python try except block. Connection Refused, Database Already Exists, Database Name Invalid, and a catch all Other Error. The current error hierarchy is enough, but it could be a lot better.

**Comment by danielmewes** [comment]
@techdragon I agree, the new error hierarchy in RethinkDB 2.1 is a first step towards making it more granular. There's a lot more to do.

**Comment by srh** [comment]
Related: #468
